### ISSUE TITLE
sFTP Access to containers

### ISSUE THREAD

**Issue by Greg-Boggs** [lando/lando#2893]
Is it possible to sFTP to the web server container created by Lando? I've tried to poke around a bit to figure out what lando ssh does, but I'm stumped. When I try to sftp to the container, I get asked for a password rather than being granted access via ssh key.

We want to try working on the site hosted in Lando without using a shared file system to see if we can improve the performance on OSX.

**Comment by Vandaahl** [comment]
Have you figured this out by any chance? I'd like to do the same thing.

**Comment by Greg-Boggs** [comment]
Nope. Even tried tweeting the lando team. we will just have to be patient I spose :)

**Comment by rubenvarela** [comment]
`lando ssh` doesn't use ssh. It does something like `docker exec -it <container> /bin/bash`

You need to enable SSH on the container. I've found some already have it and it's just starting the service. If not, you can install it. Then it's making sure the SSH port is forwarded correctly.

**Comment by pirog** [comment]
@Greg-Boggs we plan on offering a few different options for users to address slowness on macOS in Lando 4.0 which is still at least a few months or more out. That may include an SFTP option. 

In the meantime and FWIW there are several thousand people using Drupal 9 in combination with the `excludes` option which in your verbiage should make Lando "usable". See: https://docs.lando.dev/config/performance.html#performance

Also worth pointing out that Lando is not a product.

**Comment by Vandaahl** [comment]
@Greg-Boggs Please share your experience using SFTP. Does it offer the benefits you hoped it did?

**Comment by Greg-Boggs** [comment]
The excludes feature is an innovative feature. In fact, it's why I sFTP to my containers. To get good performance with Docker on OSX, you have to excludes your whole project which makes editing files difficult. That's where sFTP comes in! 

I was successful at getting good performance out of Docker on OSX and Windows. I did it by installing Linux in a VM and then Lando and Docker inside the Linux VM. It sounds complicated, but since Lando is so easy to use, it was actually pretty easy. A Linux VM more than doubled the performance of D9 in Lando on my machine.

Not to be a grammar nerd, but I do believe softwares are in fact products.

**Comment by Greg-Boggs** [comment]
Another way to get fantastic performance on Lando and Docker is to install Lando on an Amazon EC2 instance. It's super duper fast that way. But, that pretty involved because you have to set up EC2 and configure a proxy.

**Comment by pirog** [comment]
@Greg-Boggs because Lando is not for sale we consider it a project, particularly a free and open source project, and not a product. Obviously you are free to call it whatever you want though! Satisfy your inner nerd!

We are also _very excited_ for a future where Lando can run seamlessly in the cloud while still providing the same great developer experience it does today. Feel free to draw whatever conclusions you want from that statement!

**Comment by Greg-Boggs** [comment]
Providing Lando as a cloud service would be a huge feature I'd happily pay for (and support).

I suppose I refer to open source projects as products when they are a polished set of software distributed as self contained thing that people receive. While you might not sell Lando, others certainly could, and you've given us all express permission to do so in the GPL.

**Comment by pirog** [comment]
We very much think that you are not alone in that sentiment.

**Comment by Jerome-Carenity** [comment]
> Somebody figured it out, see https://github.com/beyondcode/tinkerwell-community/issues/279#issuecomment-678636276

@Vandaahl Did you by any chance know what the solution was? The link isn't working anymore. I'm struggling with an sftp connection myself and can't find anything helping

**Comment by Vandaahl** [comment]
> > Somebody figured it out, see https://github.com/beyondcode/tinkerwell-community/issues/279#issuecomment-678636276
> 
> @Vandaahl Did you by any chance know what the solution was? The link isn't working anymore. I'm struggling with an sftp connection myself and can't find anything helping

I don't remember but if you copy the url and paste it into web.archive.org I think you will find the info you want.

I gave up on docker on MacOS. Could not get it to work reliably with both speed (virtiofs) and stability (tried both Colima and Docker Desktop). I have switched to an abstraction of Nix package manager called Devbox (like Lando is an abstraction of docker compose) and now I have native speed and stability since it does not run on a virtual machine unlike Docker on MacOS.

**Comment by Jerome-Carenity** [comment]
Thank you @Vandaahl!
I didn't knew web.archive.org it's awesome. You saved my day!

**Comment by Greg-Boggs** [comment]
In my case, I switched to Windows as my host machine which uses WSL2 ubuntu to run Lando/ddev and it's wizbang fast.
